core/src/worker/mod.rs

```diff
@@ -357,6 +361,7 @@ impl Worker {
 
     /// Finish shutting down by consuming the background pollers and freeing all resources
     async fn finalize_shutdown(self) {
+        self.shutdown().await;
         if let Some(b) = self.at_task_mgr {
```

Comment:
We do this in `shutdown` already.

Reply:
Oh, you mean this bit with the task manager. This is different in some sense, it's consuming the activity manager.

Reply:
does it matter? We consume the entire worker here.